### PR TITLE
Move the Protocol version to a secret

### PR DESCRIPTION
**Pull request by andreaTP**
This change will enable having different Cloudflow versions deployed in separate namespaces in a cluster

### CODE REVIEW

File: core/cloudflow-cli/src/main/scala/akka/cli/cloudflow/kubeclient/KubeClientFabric8.scala [hunk 1]
```diff
@@ -209,18 +209,30 @@ class KubeClientFabric8(
...
+          .withLabel(App.CloudflowProtocolVersion)
+          .list()
+          .getItems()
+      }
+      protocolVersion <- {
+        protocolVersionSecret.size() match {
+          case 1 => Success(protocolVersionSecret.get(0))
+          case x if x > 1 =>
+            Failure(
+              CliException("Multiple Cloudflow operators detected in the cluster. This is not supported. Exiting"))
```
Comment (RayRoestenburg) [inline core/cloudflow-cli/src/main/scala/akka/cli/cloudflow/kubeclient/KubeClientFabric8.scala:230]: It sort of is supported, but only when using `operatorNamespace`? (maybe specify that they must use the operatorNamespace flag instead?)

File: core/cloudflow-operator/src/main/scala/cloudflow/operator/Main.scala [hunk 2]
```diff
@@ -61,6 +61,7 @@ object Main extends {
       checkCRD(settings, client)
 
       val ownerReferences = getDeploymentOwnerReferences(settings, client)
+      installProtocolVersion(settings, client, ownerReferences)
```
Comment (RayRoestenburg) [inline core/cloudflow-operator/src/main/scala/cloudflow/operator/Main.scala:64]: What about adding this to helm chart?

Comment (RayRoestenburg) [reply core/cloudflow-operator/src/main/scala/cloudflow/operator/Main.scala:64]: I guess the cloudflow currently already has permissions to create secrets in its own namespace, so is fine like this too. But feels a bit like a helm thing. (I'm ok what is done here, just a comment)

**Review by RayRoestenburg — APPROVED** [review]
One comment about feedback when more than one protocol version is found, if they did not specify the operatorNamespace, we could provide better feedback.

File: core/cloudflow-operator/src/main/scala/cloudflow/operator/Main.scala (hunk 2 above)
Comment (andreaTP) [reply core/cloudflow-operator/src/main/scala/cloudflow/operator/Main.scala:64]: make sense, I just restored the old way of doing it, for now, doing it in code makes us faster to change it, but helm makes sense too.

**Comment by RayRoestenburg** [comment]
fails on format

**Pull request merged**